bootstrap.py

```diff
@@ -0,0 +1,211 @@
+    """Attempt to push tagged builds to ghcr.io."""
+
+    push_command = ['docker', 'push', tag]
+
+    if verbose:
+        print(f'Push: {push_command}')
+
+    if dry_run:
+        return True
+
+    if not subprocess.run(push_command, stderr=None, check=False).returncode:
+        return True
+
+    print(f'Failed to push image {tag}.', file=sys.stderr)
+    return False
+
+
+def tag_image(tag, verbose=False, dry_run=False):        # pylint: disable=redefined-outer-name
+    """Tag an existing 'dunes' image with the provided tag."""
+
+    if tag is None:
+        return False
+
+    tag_command = ['docker', 'tag', 'dunes', tag]
+
+    if verbose:
+        print(f'Tagging: {tag_command}')
+
+    if dry_run is True:
+        return True
+
+    # expect a zero result for success
+    if not subprocess.run(tag_command, stderr=None, check=False).returncode:
+        return True
+
+    print(f'Failed to apply tag: {tag}', file=sys.stderr)
+    return False
+
+
+def build_image(leap_version=None, cdt_version=None, refcon_version=None, tag='dunes', verbose=False, dry_run=False):  # pylint: disable=too-many-arguments, redefined-outer-name
+    """Build a docker image for working with DUNES.
+
+    """
+
+    barg = '--build-arg'
+
+    # Set user id to a default value
+    user_id = 0
+
+    # Set group id based on platform
+    group_id = 0
+    if platform.system() == "Darwin":
+        group_id = 200
+
+    # pylint: disable=line-too-long
+    build_command = ['docker', 'build', '--no-cache', '-f','Dockerfile', '-t',tag, barg,f'USER_ID={user_id}', barg,f'GROUP_ID={group_id}', DUNES_ROOT]
+
+    if leap_version:
+        build_command.extend( [barg, f'LEAP_VERSION={leap_version}'] )
+
+    if cdt_version:
+        build_command.extend( [barg, f'CDT_VERSION={cdt_version}'] )
+
+    if refcon_version:
+        build_command.extend( [barg, f'REFCON_VERSION={refcon_version}'] )
+
+    if verbose:
+        print(f'Build command: {build_command}' )
+
+    if dry_run:
+        return True
+
+    if not subprocess.run(build_command, stderr=None, check=False).returncode:
+        return True
+
+    print('Failed to build the image.', file=sys.stderr)
+    return False
+
+
+if __name__ == "__main__":
+
+    # Create the CLI code
+    parser = argparse.ArgumentParser(
+        prog='bootstrap',
+        description='Generate a Docker image for DUNES.'
+        #epilog='Text at the bottom of help'
+    )
+
+    # Add arguments.
+    #  pylint:disable=line-too-long
+    parser.add_argument('-c', '--cdt',    dest='cdt_version',    action='store',      help='Set the leap version to CDT_VERSION.')
+    parser.add_argument('-l', '--leap',   dest='leap_version',   action='store',      help='Set the CDT version to LEAP_VERSION.')
+    parser.add_argument('--refcon',       dest='refcon_version', action='store',      help='Set the reference-contract version to REFCON_VERSION. Usually a commit hash.')
```

Comment:
Is refcon_version actually used?
I tried command ` .\bootstrap.bat --refcon main -r`
and I got:

 `[Warning] One or more build-args [REFCON_VERSION] were not consumed   `

Reply:
Right now, `--refcon` is in place for #201.
I suppressed the argument in 58af79f, but left a comment for enabling display later. I think this is a good compromise.